corehq/apps/repeaters/models.py

```diff
@@ -50,7 +50,7 @@
 def simple_post_with_cached_timeout(data, url, expiry=60 * 60, force_send=False, *args, **kwargs):
     # no control characters (e.g. '/') in keys
     key = hashlib.md5(
-        '{0} timeout {1}'.format(__name__, url)
+        u'{0} timeout {1} {2}'.format(__name__, url, data).encode('utf-8')
```

Comment:
Would be worth a discussion, actually. I'm not sure why we cache failure responses at all. 

In general repeaters currently do the following:
- If you hit any error
   - [immediately retry 3 times](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/repeaters/models.py#L589-L593) (I call this the "pummel the other server into compliance" phase) (Now that I think of it, the pummel phase is actually useless because whenever there is an error and this method is called again, it just returns the cached response.)
   - if all 3 of those fail, [requeue](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/repeaters/models.py#L605-L606) the repeater to try again in an hour

The caching *might* be useful if you hit a timeout, but there is no way of knowing whether the timeout was due to the particular data you submitted or if it is because the server is unavailable. If it is because of the data that you submitted, and we cache that timeout, we are just delaying all other requests to that server. 

I'd like to propose:
- remove the (useless) pummel phase
- remove the cache completely

This means:
- If the server is timing out because it is unavailable, we try once for each repeat record, then requeue it an hour later (then an increasing amount of time).
- If the server is timing out because of the data you submitted, it will requeue that particular record up to 3 times, then put that record into the [cancelled state](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/repeaters/models.py#L607-L609), without affecting any of the other requests to that server